src/components/FormControlEditor/FormControlEditor.tsx

```diff
@@ -10,16 +14,93 @@ editorLoader.config({
     },
 });
 
+const triggerCharacter = '@';
+const emptySuggestions = { suggestions: [] };
+
 export const FormControlEditor = React.forwardRef<HTMLDivElement, React.ComponentProps<typeof FormEditor>>(
-    ({ ...props }, ref) => (
-        <FormEditor
-            ref={ref}
-            messages={{
-                attachmentsButton: tr('Attach files'),
-                attachmentsDescription: tr("drag'n'drop or pasting also supported"),
-                attachmentsUploading: tr('Uploading...'),
-            }}
-            {...props}
-        />
-    ),
+    ({ onMount, ...props }, ref) => {
+        const utils = trpc.useContext();
+
+        const onMountCallback = useCallback<NonNullable<React.ComponentProps<typeof FormEditor>['onMount']>>(
+            (editor, monaco) => {
+                monaco.languages.registerCompletionItemProvider('markdown', {
+                    provideCompletionItems: async (model, position) => {
+                        const line = model.getValueInRange({
+                            startColumn: 0,
+                            endColumn: position.column,
+                            startLineNumber: position.lineNumber,
+                            endLineNumber: position.lineNumber,
+                        });
+
+                        if (line.indexOf(triggerCharacter) < 0) {
+                            return emptySuggestions;
+                        }
+
+                        const query = line.split(triggerCharacter).pop() || '';
```

Comment:
Maybe here to use sliced string from next  index after trigger position to line's end?

I think that transform to array is useless :)